thinc/backends/cpu_kernels.hh

```diff
@@ -226,9 +222,7 @@ void cpu_reduce_sum(A* sums__bo, const A* X__to, const L* lengths__b,
     static_assert(std::is_integral<L>::value, "Array length should be integral");
 
     for (const L* length = lengths__b; length < lengths__b + B; ++length) {
-        if (*length == 0)
-            continue;
-        else if (*length < 0)
+        if (*length < 0)
```

Comment:
Same here with the `length == 0` special case. I was just thinking that because it works differently from reduce max it might be helpful for interpretation, but maybe not.